src/System.CommandLine/Invocation/InvocationPipeline.cs

```diff
@@ -46,22 +62,53 @@ internal static async Task<int> InvokeAsync(ParseResult parseResult, Cancellatio
             {
                 terminationHandler?.Dispose();
             }
+
+            static async Task<(int returnCode, bool success)> InvokeActionAsync(ParseResult parseResult, CliAction action, CancellationToken token)
```

Comment:
We should avoid using value tuples, as they are structs so the JIT needs to perform dedicated compilation and it hurts startup scenario.

How about moving all the new logic to the existing `try/catch` blocks?

Reply:
Good call.